common/extensions/api/brave_sync.json

```diff
@@ -0,0 +1,508 @@
+          "fields" : { "type": "array", "items": {"type": "string" }, "optional": true }
+        }
+      },
+      {
+        "id": "Device",
+        "type": "object",
+        "description": "Represents sync device",
+        "properties": {
+          "name" : {"type": "string"}
+        }
+      },
+      {
+        "id": "SyncRecord",
+        "type": "object",
+        "description": "Record representing sync entity",
+        "properties": {
+          "action" : {
+            "type": "integer",
+            "description": "action code"
+          },
+          "deviceId" : {
+            "type": "binary",
+            "description": "device id"
+          },
+          "objectId" : {
+            "type": "binary",
+            "description": "object id"
+          },
+          // "deviceIdStr", "objectIdStr" and "parentFolderObjectIdStr" are
+          // workaround because js lib uses UInt8Array type for ids,
+          // and there is no appropriate conversion
+          "deviceIdStr" : {
+            "type": "string",
+            "description": "device id",
+            "optional": true
+          },
+          "objectIdStr" : {
+            "type": "string",
+            "description": "object id",
+            "optional": true
+          },
+          "objectData": {
+            "type": "string",
+            "description": "may be bookmark|device|historySite|siteSetting"
+          },
+          "syncTimestamp": {
+            "type": "number",
+            "description": "time when record was synced",
+            "optional": true
+          },
+          "bookmark" : {
+            "$ref": "Bookmark",
+            "optional": true,
+            "description": "bookmark info"
+          },
+          "historySite" : {
+            "$ref": "Site",
+            "optional": true,
+            "description": "history info"
+          },
+          "siteSetting" : {
+            "$ref": "SiteSetting",
+            "optional": true,
+            "description": "site setting"
+          },
+          "device" : {
+            "$ref": "Device",
+            "optional": true,
+            "description": "synced device info"
+          }
+        },
+        "oneOf": [
+          { "required": [ "bookmark"] },
+          { "required": [ "history_site"] },
+          { "required": [ "site_setting"] },
+          { "required": [ "device"] }
+        ]
+      },
+      {
+        "id": "RecordAndExistingObject",
+        "type": "object",
+        "description": "represents pair of sync record and sync record with existing object",
+        "properties": {
+          "serverRecord" : {
+            "$ref": "SyncRecord",
+            "description": "original sync record"
+          },
+          "localRecord" : {
+            "$ref": "SyncRecord",
+            "description": "existing sync record",
+            "optional": true
+          }
+        }
+      }
+    ],
+    "events": [
+      {
+        "name": "onBrowserToBackgroundPage",
```

Comment:
is this still just leftover code that hasn't been migrated to more specific events?